src/main/java/kervyn/Tasks/TaskList.java

```diff
@@ -0,0 +1,241 @@
+            Task task = userTasks.get(Integer.parseInt(processedUserInput[1]) - 1);
+            if (task.getStatus()) {
+                taskAlreadyMarked(dialogContainer, kervynImage);
+            } else {
+                dialogContainer.getChildren().add(
+                        DialogBox.getKervynDialog("\tNice! I've marked this task as done:", kervynImage)
+                );
+                task.updateStatus();
+                dialogContainer.getChildren().add(
+                        DialogBox.getKervynDialog(task.toString(), kervynImage)
+                );
+            }
+
+            return 1;
+        }
+        catch (IndexOutOfBoundsException e) {
+            // Need to account for trying to mark a task that doesn't exist
+            dialogContainer.getChildren().add(
+                    DialogBox.getKervynDialog("\tTask number provided doesn't exist. Please try again.", kervynImage)
+            );
+        }
+        return 0;
+    }
+
+    /**
+     * Unmarks a task as not completed in the task list.
+     *
+     * @param userTasks The ArrayList of Task objects.
+     * @param processedUserInput The user input processed into an array of Strings.
+     * @return Returns 1 if the unmark operation was successful, 0 otherwise.
+     */
+    public short unMarkTask(ArrayList<Task> userTasks, String[] processedUserInput, Image kervynImage, VBox dialogContainer) {
+        try {
+            Task task = userTasks.get(Integer.parseInt(processedUserInput[1]) - 1);
+            if (!task.getStatus()) {
+                taskAlreadyUnMarked(dialogContainer, kervynImage);
+            } else {
+                dialogContainer.getChildren().add(
+                        DialogBox.getKervynDialog("\tOK, I've marked this task as not done yet:", kervynImage)
+                );
+                task.updateStatus();
+                dialogContainer.getChildren().add(
+                        DialogBox.getKervynDialog(task.toString(), kervynImage)
+                );
+            }
+
+            return 1;
+        }
+        catch (IndexOutOfBoundsException e) {
+            // Need to account for trying to unmark a task that doesn't exist
+            dialogContainer.getChildren().add(
+                    DialogBox.getKervynDialog("\tTask number provided doesn't exist. Please try again.", kervynImage)
+            );
+        }
+        return 0;
+    }
+
+    /**
+     * Private method to handle the scenario when a task is already marked.
+     */
+    private void taskAlreadyMarked(VBox dialogContainer, Image kervynImage) {
+        dialogContainer.getChildren().add(
+                DialogBox.getKervynDialog("\tUh oh! It looks like this task is already marked as done, please try again with another task!", kervynImage)
+        );
+    }
+
+    /**
+     * Private method to handle the scenario when a task is already unmarked.
+     */
+    private static void taskAlreadyUnMarked(VBox dialogContainer, Image kervynImage) {
```

Comment:
Similarly, can consider using a verb name for the method, e.g. `handleUnmarkedTask`.